lib/accesskey.js

```diff
@@ -0,0 +1,179 @@
+ */
+
+var crypto = require('crypto');
+var crc32 = require('crc').buffer.crc32;
+var assert = require('assert-plus');
+
+var TO_B64_REG = new RegExp('[+/=]', 'g');
+var FROM_B64_REG = new RegExp('[-_]', 'g');
+
+var DEFAULT_PREFIX = 'tdc_';
+var DEFAULT_BYTE_LENGTH = 32;
+
+// Don't have base64url encoded Buffers until Node v14
+function toBase64url(input) {
+  return input
+    .toString('base64')
+    .replace(TO_B64_REG, function (c) {
+        if (c === '+') {
+          return '-';
+        }
+        if (c === '/') {
+          return '_';
+        }
+        if (c === '=') {
+          return '';
+        }
+        return null;
+    });
+}
+
+function fromBase64url(input) {
+  var base64 = input.replace(FROM_B64_REG, function (c) {
+    if (c === '-') {
+      return '+';
+    }
+    if (c === '_') {
+      return '/';
+    }
+    return null;
+  });
+
+  // Restore padding
+  while (base64.length % 4 !== 0) {
+    base64 += '=';
+  }
+
+  return Buffer.from(base64, 'base64');
+}
+
+/**
+ * Generate a random secret access key inspired by suggestions from Github's
+ * Secret Scanning Partner Program[0] and how they structure their keys[1]:
+ *   [0] https://i.no.de/c12f50d544eececf
+ *   [1] https://i.no.de/5a4e8cea87c0a873
+ *
+ * Instead of using Base62 as Github does, base64url encoding is used instead.
+ *
+ * Keys generated from this function have:
+ *   - A uniquely defined prefix (e.g. "tdc_" for "Triton DataCenter")
+ *   - High entropy random strings (32 random bytes from node crypto)
+ *   - A 32-bit crc checksum (to validate token structure)
+ *
+ * An example key:
+ *
+ *    tdc_SU4xWXL-HzrMIDM_A8GH94sl-uc-aX8mqsEMiK4JSVdAGyjH
+ *
+ *    +--------+--------------------------------------------+--------+
+ *    | PREFIX |             RANDOM BYTES                   | CRC32  |
+ *    +--------+--------------------------------------------+--------+
+ *    |  tdc_  | SU4xWXL-HzrMIDM_A8GH94sl-uc-aX8mqsEMiK4JSV | dAGyjH |---+
+ *    +--------+--------------------------------------------+--------+   |
+ *             |                 BASE64 URL ENCODED                  |   |
+ *    +--------+--------------------------------------------+--------+   |
+ *    |       CRC32 coverage (PREFIX + RANDOM BYTES)        | <----------+
+ *    +-----------------------------------------------------+
+ *
+ * @param {String} prefix string for the token.
+ * @param {Number} byte count to randomly generate.
+ * @param {Function} callback of the form fn(err, key).
+ * @throws {TypeError} on bad input.
+ */
+function generate(prefix, bytes, done) {
+    assert.string(prefix, 'prefix');
+    assert.number(bytes, 'bytes');
+    assert.func(done, 'done');
+
+    crypto.randomBytes(bytes, function generateBytes(err, randBytes) {
+        if (err) {
+            done(err);
+            return;
+        }
+
+        // Create a buffer containing the prefix and random bytes
+        var prefixBuf = Buffer.alloc(prefix.length, prefix);
```

Comment:
As we discussed previously, if possible, we need to avoid using `Buffer.alloc` even if the package.json engines is specified as :
 ```
"engines": {
    "node": ">=0.10.48"
  },
```
`Buffer.alloc` is not present in Node.js 0.10.48, which is the version that Mahi uses, and currently, Mahi is a consumer for node-ufds.

Reply:
All is working fine after that change!.